sdks/java/extensions/join-library/README.md

```diff
@@ -46,7 +46,7 @@ Join-library can be found on maven-central:
     <dependency>
       <groupId>org.apache.beam</groupId>
       <artifactId>join-library</artifactId>
-      <version>0.1-incubating-SNAPSHOT</version>
```

Comment:
This whole section seems bad. I think we only had it because it was not distributed as part of DataflowJavaSDK. Now that it's part of Beam, drop this block?